src/components/__tests__/utils.test-d.ts

```diff
@@ -0,0 +1,28 @@
+import { expectType } from 'tsd'
+import { CountryData } from '../..'
+import utils from '../../components/utils'
+
+// Expect the predicate to do a matching operation on each item in the array of generic items passed to the first argument utils.findIndex().
+// e.g. item is CountryData, (items: CountryData, predicate: (item: CountryData) => boolean) => number
+type FindIndexFunction<F extends <I>(items: I[], predicate: (item: I) => boolean | undefined) => number> = F
+expectType<FindIndexFunction<typeof utils.findIndex>>(utils.findIndex)
+
+const countries: CountryData[] = []
+const countryCode: string | undefined = undefined
+const ignoreOnlyCountriesOption: boolean | undefined = undefined
+
+// Expect null to never be returned when `allowFail` is false in utils.getCountryData().
+expectType<CountryData>(utils.getCountryData(
+  countries,
+  countryCode,
+  ignoreOnlyCountriesOption,
+  false
+))
+
+// Expect null to be returned conditionally when `allowFail` is true in utils.getCountryData().
+expectType<CountryData | null>(utils.getCountryData(
+  countries,
+  countryCode,
+  ignoreOnlyCountriesOption,
+  true
+))
```

Comment:
I'm doubting the need to use [`tsd`](https://github.com/SamVerschueren/tsd) yet. See `expectType` in [the **Usage** section](https://github.com/SamVerschueren/tsd#usage).

The reason I chose to have `tsd` is solely because [`findIndex` in **utils.js**](https://github.com/patw0929/react-intl-tel-input/blob/v8.0.5/src/components/utils.js#L174-L186) has a type-flexible `predicate` callback argument. My principle of battle-testing APIs inside out is showing here but I think that having `tsd` for just this one function is starting to look like overkill.

What say you I cut `tsd` out and let `tsc` do the heavy lifting for now? @mcataford

Reply:
Sure, that sounds good!

Reply:
#392

Reply:
@mcataford #392 will resolve this.